app/controllers/time_slot_controller.rb

```diff
@@ -68,6 +68,8 @@ def index
           appointments = Appointment.where(start: @current_day.at_beginning_of_day..@current_day.end_of_day, ubs: ubs, patient_id: nil)
         end
 
+        next unless appointments.present?
```

Comment:
O `.present?` carrega todos os registros e verifica se o array tem pelo menos um item. Melhor usar `.exists?`, que otimiza a query pra determinar apenas se existe um registro.

```suggestion
        next unless appointments.exists?
```